Bank

```diff
@@ -0,0 +1,183 @@
+pragma solidity ^0.4.16;
+
+import './Ownable.sol';
+import './SafeMath.sol';
+
+contract Bank is Ownable{
+    using SafeMath for uint256;
+    
+    uint BankTotal;
+    uint minDeposit;
+    uint minWithdraw;
+    uint maxWithdraw;
+    uint DepositBonus;
+    uint transferTax;
+    
+    //mapping
+    mapping (address=> Account)public Accounts;
+    
+    //enum
+   enum accountType {saving,current,premium}
+
+   
+    
+    //events
+    event EDeposit(address a,uint b);
+    event Ewithdraw(address a,uint b);
+    event ETransfer(address sender,address receiver,uint c);
+    
+    // Account Struct
+    struct Account{
+        string name;
+        address _address;
+        uint ID;
+        uint balance;
+        bool open;
+        accountType Type;
+        
+    }
+    
+    modifier onlyUser()
+    {
+        require(Accounts[msg.sender].open == true);
+        _;
+    }
+    
+//getters
+    
+        function getBalance()public view returns(uint)
+        {
+            return Accounts[msg.sender].balance;
+            
+        }
+        
+        function getBankTotal()onlyOwner public view returns(uint)
+        {
+            return BankTotal;
+        }
+ 
+        
+ //setters
+
+        function setMinDeposit(uint _value)onlyOwner public
```

Comment:
validate arguments before assigning

Reply:
we are validating arguments in deposit,withraw and transfer function already, should we validate values in setters which are onlyOwner already?